### PR TITLE
Pass comments to CDC events

### PR DESCRIPTION
**Pull request by kgalieva**
## Summary
- Extract SQL comments (`/* ... */`) at query parse time in the PostgreSQL layer and propagate them through the write path into CDC change events
- Works for both single-shard and transactional (multi-shard) writes
- Enables CDC consumers to correlate change events with application-level context (e.g., tenant ID, pod ID, request ID) embedded in SQL comments

## Motivation

When consuming CDC events from YugabyteDB, there is no way to correlate a change record back to the application context that produced it. This is especially important for multi-tenant systems where a single table serves multiple tenants — CDC consumers need to know which tenant a write belongs to.

SQL comments (`/* pod_id=42*/`) are a natural, non-intrusive way to attach metadata to queries. This PR makes those comments available in CDC output so downstream consumers can filter, route, or tag events accordingly.

## Feature flag

The entire feature is gated behind the GFlag `cdc_propagate_query_comments` (default `false`):
- **PG layer**: `extract_first_sql_comment()` returns NULL immediately when flag is false
- **Write path**: `write_query.cc` skips copying `query_comment` into the Raft `WritePB` when flag is false
- **Result**: zero WAL overhead and zero protobuf overhead on the Raft write path when the feature is disabled

## Limitations

- The comment must appear **before** the SQL statement: `/* pod_id=42 */ INSERT INTO ...`. Only leading comments (after optional whitespace) are extracted. Mid-query or trailing comments are ignored.

- **Comment size limit**: extracted comments are truncated to 1024 bytes to prevent unbounded WAL bloat.

- **One comment per transaction.** For transactional writes, the comment from the first WRITE_OP in the WAL is used for all records in that transaction. This is correct for the target use case where all statements in a request share the same application context (pod_id, shop_id, etc.).

- **Large transactions spanning GetChanges batches.** The `txn_query_comments` map is scoped to a single `GetChanges` call. If a transaction's intent WRITE_OP is in one batch and the APPLY record is in a later batch (very large transactions exceeding the response size), the comment may be missing from the later batch's records. In practice, the consistent WAL reader batches WRITE_OPs and APPLYs together by safe time, and the non-consistent reader with `skip_intents=false` keeps intents in the batch. This edge case only affects partially-streamed large transactions.

### Single-shard writes (non-transactional)
1. **Parse time** (`parser.c`): `extract_first_sql_comment()` extracts the content between `/*` and `*/` from the beginning of the query string (gated by flag)
2. **PG executor** (`nodeModifyTable.c`, `ybModifyTable.c`, `pg_yb_utils.c`): The extracted comment is passed to pggate via `YBCPgNewInsert`/`Update`/`Delete` as a new `query_comment` parameter
3. **PgGate** (`pg_dml_write.cc`): Stores the comment on the write request and copies it into `PgsqlWriteRequestPB.query_comment` (field 30)
4. **Write path** (`write_query.cc`): Copies `query_comment` from the first PgsqlWriteRequestPB into the Raft `WriteRequestPB` (field 23 in `operations.proto`) — gated by flag
5. **CDC producer** (`cdcsdk_producer.cc`): `PopulateCDCSDKWriteRecord` reads `query_comment` from the WAL entry and sets it on `RowMessage.query_comment` (field 20 in `cdc_service.proto`)

### Transactional writes
1. Steps 1-4 are the same — the `query_comment` is written to the WAL in the intent WRITE_OP
2. **WAL scan** (`GetConsistentWALRecords`/`GetWALRecords`): Before skipping intent WRITE_OPs, the CDC producer extracts `query_comment` and stashes it in a `txn_query_comments` map keyed by `TransactionId`
3. **Intent processing**: When the APPLY record triggers `ProcessIntents` → `PopulateCDCSDKIntentRecord`, the stashed comment is looked up by transaction ID and set on each `RowMessage`

## Files changed (feature-specific)

| Layer | Files | What |
|-------|-------|------|
| Proto | `cdc_service.proto`, `pgsql_protocol.proto`, `operations.proto` | Added `query_comment` fields |
| Parser | `parser.c`, `parser.h` | `extract_first_sql_comment()` with position validation and size cap |
| PG executor | `nodeModifyTable.c`, `ybModifyTable.c`, `pg_yb_utils.c/h` | Pass comment to pggate |
| PgGate | `ybc_pggate.h/cc`, `pg_dml_write.h/cc`, `pggate.cc/h` | New parameter on Insert/Update/Delete/InsertBlock |
| GFlags | `pggate_flags.cc` (DEFINE), `ybc_gflags.h/cc`, `cdcsdk_producer.cc`, `write_query.cc` (DECLARE) | `cdc_propagate_query_comments` flag |
| Write path | `write_query.cc` | Copy comment into Raft write request (gated by flag) |
| CDC producer | `cdcsdk_producer.cc` | Emit comment in CDC `RowMessage`; `txn_query_comments` map uses `TransactionId` key |
| Tests | `cdcsdk_ysql-test.cc`, `pggate_test_*.cc` | Integration tests including edge cases and flag-disabled test |

## Test plan

- [x] Single-shard INSERT with comment: `query_comment` appears in CDC event
- [x] Single-shard INSERT without comment: no `query_comment` field in CDC event
- [x] Transactional INSERT (`BEGIN`/`COMMIT`) with comment: `query_comment` appears in CDC event
- [x] Transactional UPDATE (`BEGIN`/`COMMIT`) with comment: `query_comment` appears in CDC event
- [x] Flag disabled: no `query_comment` in CDC events even with comment in SQL
- [x] Edge cases: empty comment, trailing comment, mid-query comment — all correctly excluded
- [x] Single-shard DELETE and UPDATE with comment: `query_comment` appears in CDC event
- [x] Multi-tablet transactional writes: `query_comment` propagated via intent/APPLY path across tablets
- [x] C++ integration tests: `QueryCommentSingleShard`, `QueryCommentTransactional`, `QueryCommentDisabledByFlag`, `QueryCommentEdgeCases`, `QueryCommentDeleteAndUpdate`, `QueryCommentMultiTabletTransaction`

---

> [!NOTE]
> **Medium Risk**
> Touches the YSQL write path, protobuf/WAL payloads, and CDC WAL processing; although gated by a default-off flag, enabling it changes replication output and adds per-write metadata that could impact WAL/CDC behavior if buggy.
> 
> **Overview**
> Adds an **optional SQL `query_comment` field to CDC records** and plumbs it end-to-end from YSQL statements into CDC output.
> 
> When the new runtime flag `cdc_propagate_query_comments` is enabled, the PG layer extracts the first leading `/* ... */` comment (trimmed and capped) and passes it through updated pggate statement constructors into `PgsqlWriteRequestPB`/raft `WritePB`, and the CDC producer emits it on `RowMessage` for both single-shard writes and distributed transactions (by stashing intent comments per `TransactionId` during WAL scans and reusing them when processing APPLY/intents). New integration tests validate inserts/updates/deletes, transactional vs single-shard behavior, edge cases, and flag-off behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b2621b753ad60ff8f05d9fc35c4de4823f3a4068. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/postgres/src/backend/parser/parser.c [hunk 1]
```diff
@@ -499,3 +499,59 @@ str_udeescape(const char *str, char escape,
...
+	char	   *comment;
+	char	   *p;
+
+	if (!query_string)
+		return NULL;
+
+	/* Find start of comment */
+	start = strstr(query_string, "/*");
+	if (!start)
+		return NULL;
```
[the remark this answers is not in the document]
Comment (kgalieva) [reply src/postgres/src/backend/parser/parser.c:538]: Added a Limitations section explaining the comment-before-query requirement

File: src/postgres/src/backend/parser/parser.c [hunk 2]
```diff
@@ -499,3 +502,117 @@ str_udeescape(const char *str, char escape,
...
+		/* Must be the start of a block comment */
+		if (p[0] != '/' || p[1] != '*')
+			return NULL;
+
+		content = p + 2;
+
+		/* Find end of comment */
+		end = strstr(content, "*/");
+		if (!end)
+			return NULL;
```
[the remark this answers is not in the document]
Comment (kgalieva) [reply src/postgres/src/backend/parser/parser.c:561]: It is described in the limitations: 

> The comment must appear before the SQL statement: /* pod_id=42 */ INSERT INTO .... Only leading comments (after optional whitespace) are extracted. Mid-query or trailing comments are ignored.
